### PR TITLE
Add helm chart

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/15992

### CODE REVIEW

File: helm/cluster-api/templates/role.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-leader-election-role
```
Comment (MarcelMue) [inline helm/cluster-api/templates/role.yaml:6]: Do we need this? I removed all leader election stuff from my side

File: helm/cluster-api-control-plane/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,184 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-aggregated-manager-role
```
Comment (MarcelMue) [inline helm/cluster-api-control-plane/templates/rbac.yaml:6]: I think we don't need this one - only the `manager-role`

File: helm/cluster-api-control-plane/templates/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-manager-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ include "resource.default.name" . }}-aggregated-manager-role
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (MarcelMue) [inline helm/cluster-api-control-plane/templates/rbac.yaml:168]: We should create a proper service account and bind to that.

Comment (kubasobon) [reply helm/cluster-api-control-plane/templates/rbac.yaml:168]: I've seen that part in your PR, I'll copy the approach.

File: helm/cluster-api-control-plane/templates/rbac.yaml (hunk 2 above)
Comment (kubasobon) [reply helm/cluster-api-control-plane/templates/rbac.yaml]: Removed

File: pkg/project/project.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package project
```
Comment (TheoBrigitte) [inline pkg/project/project.go:1]: This file is not needed. Please remove it.

File: helm/cluster-api-control-plane/templates/rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-proxy-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ include "resource.default.name" . }}-proxy-role
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (TheoBrigitte) [inline helm/cluster-api-control-plane/templates/rbac.yaml:113]:
```suggestion
  name: {{ include "resource.default.name" . }}
```

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**